cfme/tests/candu/test_utilization_metrics.py

```diff
@@ -302,3 +316,90 @@ def test_raw_metric_host_disk(metrics_collection, appliance, provider):
         if record.disk_usage_rate_average is not None:
             assert record.disk_usage_rate_average > 0, 'Zero Host Disk IO'
             break
+
+
+def query_metric_rollup_table(appliance, provider, metric, azone_name):
+    metrics_tbl = appliance.db.client['metric_rollups']
+    ems = appliance.db.client['ext_management_systems']
+
+    with appliance.db.client.transaction:
+        provs = (
+            appliance.db.client.session.query(metrics_tbl.id)
+            .join(ems, metrics_tbl.parent_ems_id == ems.id)
+            .filter(metrics_tbl.resource_name == azone_name,
+            ems.name == provider.name)
+        )
+    return appliance.db.client.session.query(metrics_tbl).filter(
+        metrics_tbl.id.in_(provs.subquery()))
+
+
+def generic_test_azone_rollup(appliance, provider, metric):
+    azone_name = provider.data["cap_and_util"]["capandu_azone"]
+    query = query_metric_rollup_table(appliance, provider, metric, azone_name)
+
+    for record in query:
+        if hasattr(record, metric):
+            assert getattr(record, metric) > 0, (
+                'Zero Azone CPU usage/memory usage/Disk IO/Network IO'
+            )
+            break
+
+
+@pytest.mark.provider(
+    [EC2Provider, AzureProvider],
+    required_fields=[(['cap_and_util', 'capandu_vm'], 'cu-24x7')],
+    override=True, scope="module"
+)
+@pytest.mark.meta(
+    blockers=[BZ(1744845, forced_streams=['5.10'],
+        unblock=lambda provider: not provider.one_of(AzureProvider))]
+)
+def test_azone_cpu_usage(metrics_collection, appliance, provider):
```

Comment:
Consider parametrizing here, as it looks like all three test functions have the same provider marks, same BZ blocking, same polarion meta, and same function call, just with a different string for the rollup metric.

Could just parametrize those metric names and have a single function, so all these provider marks aren't repeated so much.

That or wrap them in a class at least so you can apply the provider mark to the class.

Reply:
@mshriver https://github.com/ManageIQ/integration_tests/pull/9530#discussion_r339027325